libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/FileAccessTree.java

```diff
@@ -139,10 +141,13 @@ private FileAccessTree(
             });
         }
 
-        // everything has access to the temp dir, config dir and the jdk
+        // everything has access to the temp dir, config dir, to their own dir (their own jar files) and the jdk
         addPathAndMaybeLink.accept(pathLookup.tempDir(), READ_WRITE);
         // TODO: this grants read access to the config dir for all modules until explicit read entitlements can be added
         addPathAndMaybeLink.accept(pathLookup.configDir(), Mode.READ);
+        if (componentDir != null) {
```

Comment:
If we change to always at least allow access to a module's jar itself, then this should never be null?

Reply:
It depends; if we are always OK with doing `Paths.get(requestingClass.getProtectionDomain().getCodeSource().getLocation().toURI())` then yes, this will never be null. If we are not OK with that, there are cases (unknown modules, agents) in which this might still be null.
IDK about this; what about Painless code for example; they do have a class, but do they have a "jar"? (would that URL/path resolve to anything sensible?)

Reply:
Painless code doesn't have a jar, it's just a class compiled in memory.

For the others, we should still have a jar location, even for agents. In `PolicyManager.computeEntitlements` we can extract the codesource path and pass that to the construction of FileAccessTree?

Reply:
Ah, but for painless it will be null, you are right. And javadocs say getCodesource() can return null.